### ISSUE TITLE
Incorrect CFG from structured statement

### ISSUE THREAD

**Issue by shazqadeer** [boogie-org/boogie#246]
Consider the program below.

```
procedure Enter()
{
    var _old, curr: int;
    while (true) {
        call _old := CAS(0, 1);
        if (_old == 0) {
            break;
        }
        while (true) {
            call curr := Read();
            if (curr == 0) {
                break;
            }
        }
    }
}

procedure CAS(x:int, y:int) returns (z: int);
procedure Read() returns (z: int);

````
I expect to see two loops in the CFG of Enter, one nested inside the other.  However, I see three loops in the printed CFG.  Here is the output obtained using /printCFG:

digraph G {
  "anon0" [shape=box];
  "anon6_LoopHead" [shape=box];
  "anon6_LoopDone" [shape=box];
  "anon6_LoopBody" [shape=box];
  "anon7_Then" [shape=box];
  "anon7_Else" [shape=box];
  "anon3" [shape=box];
  "anon8_LoopHead" [shape=box];
  "anon8_LoopDone" [shape=box];
  "anon8_LoopBody" [shape=box];
  "anon9_Then" [shape=box];
  "anon9_Else" [shape=box];
  "anon0" -> "anon6_LoopHead";
  "anon6_LoopHead" -> "anon6_LoopDone";
  "anon6_LoopHead" -> "anon6_LoopBody";
  "anon6_LoopBody" -> "anon7_Then";
  "anon6_LoopBody" -> "anon7_Else";
  "anon7_Else" -> "anon3";
  "anon3" -> "anon8_LoopHead";
  "anon8_LoopHead" -> "anon8_LoopDone";
  "anon8_LoopHead" -> "anon8_LoopBody";
  "anon8_LoopBody" -> "anon9_Then";
  "anon8_LoopBody" -> "anon9_Else";
  "anon9_Else" -> "anon8_LoopHead";
  "anon9_Then" -> "anon0";
  "anon8_LoopDone" -> "anon6_LoopHead";
}

**Comment by bkragl** [comment]
For convenience, here is a rendering of the above CFG description.

![Enter](https://user-images.githubusercontent.com/6729391/86741599-cd667680-c037-11ea-9298-9ba378efcdf2.png)

**Comment by shazqadeer** [comment]
The situation is even worse.  Consider the following variation of the original program:
```

procedure Enter()
{
    var _old, curr: int;
    _old := 42;
    while (true) {
        call _old := CAS(0, 1);
        if (_old == 0) {
            break;
        }
        while (true) {
            call curr := Read();
            if (curr == 0) {
                break;
            }
        }
    }
}

procedure CAS(x:int, y:int) returns (z: int);
procedure Read() returns (z: int);
```

The break in the nested while loop is translated as a jump to the control location before the assignment  _old := 42.  

Upon debugging the translation of the desugaring of structured statements into CFG, it seems that this bug pops up when a nested while loop, say X, does not have any statement following it causing the continuation of X to be incorrectly determined as the enclosing big block which, in this case is the block beginning with _old  := 42.  I have attempted a fix in PR #247.
